pkg/telemetry/datadog.go

```
@@ -0,0 +1,82 @@
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u, nil)
+	if err != nil {
+		return Result{}, err
+	}
+	req.Header.Set("DD-API-KEY", p.apiKey)
+	req.Header.Set("DD-APPLICATION-KEY", p.appKey)
+
+	resp, err := p.client.Do(req)
+	if err != nil {
+		return Result{}, fmt.Errorf("datadog query: %w", err)
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)